Adding feature to ribocounts that saves non-rRNA read IDs

lennijusten opened the pull request:
RiboDetector takes a long time to run and we're currently only saving the number of rRNA reads. If we want to know the rRNA status of a read, we'd have to run it through RiboDetector again.  I expect this information to be useful in the future, and it's of very little cost to save the read IDs.

Here, I add a feature that saves a text file of non-rRNA read IDs in a sample, with the IDs parsed by `FastqGeneralIterator`. The text file is saved to an AWS directory within each bioproject called `ribopass-reads/`.

Question: Is it worth compressing the text files before copying them to AWS?

- Update: I'm now compressing txt files before copying them to AWS

jeffkaufman commented on README.md:
```diff
@@ -101,11 +101,16 @@ Available files, and their formats:
      * Plus a `settings` file with info on how cleaning went
 
-* `ribocounts/`: Intermediate, processed output of `RiboDetector`
+* `ribocounts/`: Intermediate, counts of `RiboDetector` output
```
Is it worth keeping this?  Since we can easily get the count by just checking the number of lines in a decompressed `ribopass-reads/` file.

lennijusten replied:
I'm fine removing them. Do you have a take on whether we should save the read IDs for non-rRNA reads or for rRNA reads? 

Currently I'm saving read-IDs of non-rRNA sequences since this is the default output of RiboDetector. 

When we want to display % rRNA reads on the dashboard I will need to combine this with number of total reads.

jeffkaufman replied:
No preference on which set of read ids to keep; it's easy enough to get the full list of read ids and compute the complement.

Combining with the total number of reads is also not hard.

lennijusten commented:
@jeffkaufman I updated the RiboDetector feature from a `ribocounts()` to `riboreads()` paradigm. The new function saves read titles for rRNA reads to AWS in a directory called `riboreads/`. I'm starting to re-run the bioprojects now. 

All AWS directories titled `ribocounts/` or `ribopass-reads/` can be deleted. 

I also updated the `prepare_dashboard` files to the new paradigm as well, which means the `dashboard/ribocounts/` dir can be deleted. Running `prepare_dashboard.sh` will pull files into a new dir called `dashboard/riboreads/`.

jeffkaufman commented on run.py:
```diff
@@ -255,52 +252,83 @@ def calculate_average_read_length(file_path):
+            # Gzip the text file
+            with open(riboreads_file, 'rb') as orig_file:
+                with gzip.open(gzipped_file_path, 'wb') as gzipped_file:
+                    gzipped_file.writelines(orig_file)
```
You can do these two in a single pass:

```
with gzip.open(gzipped_file_path, 'wb') as gzipped_file:
    for title in sample_rrna_reads:
        gzipped_file.write(title + '\n')
```
